src/common/buffer.cc

```diff
@@ -125,10 +128,12 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     if (r < 0)
       return r;
     buf[r] = '\0';
-    size_t size = strict_strtol(buf, 10, &err);
-    if (!err.empty())
-      return -EIO;
-    buffer_max_pipe_size.set(size);
+    errno = 0;
+    size_t size = strtoul(buf, nullptr, 10);
+    if (errno == 0)
```

Comment:
please note that errno is not aways set if `strtoul()` does not parse any character. better off passing and `endptr` to it and check if it is equal to buf. and then check the return value, and `errno` maybe.